benchmarks/time/src/main/scala/strawman/collection/immutable/ArrayBaselineBenchmark.scala

```diff
@@ -0,0 +1,148 @@
+package strawman.collection.immutable
+
+import java.util.concurrent.TimeUnit
+
+import org.openjdk.jmh.annotations._
+import org.openjdk.jmh.infra.Blackhole
+import strawman.collection.{ArrayView, View}
+import scala.{Any, AnyRef, Int, Long, Unit, Array}
+import scala.Predef.intWrapper
+
+@BenchmarkMode(scala.Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Fork(15)
+@Warmup(iterations = 30)
+@Measurement(iterations = 15)
+@State(Scope.Benchmark)
+class ArrayBaselineBenchmark {
+
+  @Param(scala.Array("39", "282", "73121", "7312102"))
+  var size: Int = _
+
+  @Param(scala.Array("39"))
+  var vLoSize: Int = _
+
+  var shortRangingFactor : Int = (size * 0.2).toInt
+
+  var v: Array[Long] = _
+  var vLo: Array[Long] = _
+
+  @Setup(Level.Trial)
+  def initData(): Unit = {
+
+    def fillArray(range: Int) = {
+      val array = new Array[Long](range)
+      var i = 0
+      while (i < range) {
+        array(i) = scala.util.Random.nextInt(size).toLong
+        i += 1
+      }
+      array
+    }
+
+    v = fillArray(size)
+    vLo = fillArray(vLoSize)
+  }
+
+  @Benchmark
+  def sum(bh: Blackhole) = {
+    var i = 0
+    var ret = 0L
+    while (i < v.length) {
+      ret += v(i)
+      i += 1
+    }
+    bh.consume(ret)
+  }
+
+  @Benchmark
+  def sumOfSquares(bh: Blackhole) = {
+    var i = 0
+    var ret = 0L
+    while (i < v.length) {
+      ret += v(i) * v(i)
+      i += 1
+    }
+    bh.consume(ret)
+  }
+
+  @Benchmark
+  def sumOfSquaresEven(bh: Blackhole) = {
+    var i = 0
+    var ret = 0L
+    while (i < v.length) {
+      if (v(i) % 2L == 0L)
+        ret += v(i) * v(i)
+      i += 1
+    }
+    bh.consume(ret)
+  }
+
+  @Benchmark
+  def maps(bh: Blackhole)= {
+    var i=0
+    var ret=0L
+    while (i < v.length) {
+      ret += v(i) +
+        (v(i) & 0xD) + 0xCAFED00D +
```

Comment:
This test is misleading since it looks like the additions will be separate.  The compiler ought to merge the constants--so either use multiplication between lines, or manually do the addition.  (The answer is `0xB2D9676B`.)